File: ccif/src/java/other/org/himalay/ccif/DviStatusReport.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:56 EST 2016")

public  class DviStatusReport extends BinStruct implements PublicBinMsg {




    // members variables
    // bf1
    public BitField_8 bf1 ;
    // videoFormat
    public VideoFormat videoFormat ;

    public DviStatusReport () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize bf1
        bf1 = new BitField_8();
        // Initialize videoFormat
        videoFormat = new VideoFormat();
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read bf1
        retVal += bf1.read(istream);
        // read videoFormat
        retVal += videoFormat.read(istream);

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;


        {   /** fix dependent sizes for videoFormat **/
        }

        // write bf1
        ostream.writeByte(bf1.getValue());
        retVal +=1;
        // write videoFormat
        if (videoFormat!=null)retVal +=videoFormat.write(ostream);
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("DviStatusReport\n");
        dc.increaseIndent();
        int retVal= 0;
        // write bf1
        {
            dc.indent();
            dc.getPs().print("reserved: ");
            dc.getPs().println(BitField_8.toDisplayString(getReserved(),3));
            dc.indent();
            dc.getPs().print("connectionStatus: ");
            dc.getPs().println(BitField_8.toDisplayString(getConnectionStatus(),2));
            dc.indent();
            dc.getPs().print("hostHdcpStatus: ");
            dc.getPs().println(BitField_8.toDisplayString(getHostHdcpStatus(),1));
            dc.indent();
            dc.getPs().print("deviceHdcpStatus: ");
            dc.getPs().println(BitField_8.toDisplayString(getDeviceHdcpStatus(),2));
        }
        // write videoFormat
        if ( videoFormat != null ) {
            dc.indent();
            dc.getPs().println("videoFormat") ;
            retVal +=videoFormat.dump(dc);
        }
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for bf1
    //public BitField_8 getBf1()
    //{
    //    return bf1 ;
    //}


    // Setter for bf1
    //public void setBf1(BitField_8 val)
    //{
    //    this.bf1= val;
    //}
    // Getter for videoFormat
    //public VideoFormat getVideoFormat()
    //{
    //    return videoFormat ;
    //}


    // Setter for videoFormat
    //public void setVideoFormat(VideoFormat val)
    //{
    //    this.videoFormat= val;
    //}


    public int getReserved( )
    {
        return ( bf1.getValue() & 0x000000e0 ) >> 5 ;
    }


    public void setReserved(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xffffff1f) | ( (val << 5 ) & 0x000000e0));
    }


    public int getConnectionStatus( )
    {
        return ( bf1.getValue() & 0x00000018 ) >> 3 ;
    }


    public void setConnectionStatus(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xffffffe7) | ( (val << 3 ) & 0x00000018));
    }


    public int getHostHdcpStatus( )
    {
        return ( bf1.getValue() & 0x00000004 ) >> 2 ;
    }


    public void setHostHdcpStatus(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xfffffffb) | ( (val << 2 ) & 0x00000004));
    }


    public int getDeviceHdcpStatus( )
    {
        return ( bf1.getValue() & 0x00000003 ) >> 0 ;
    }


    public void setDeviceHdcpStatus(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xfffffffc) | ( (val << 0 ) & 0x00000003));
    }


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/DviStatusReport.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:13 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "DviStatusReport.h"

#ifdef org_himalay_ccif_DviStatusReport__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
DviStatusReport::DviStatusReport() {
    references= 0;


    // bf1
    bf1 = 0;
    // videoFormat
    org_himalay_ccif_VideoFormat__NEW( videoFormat);
}
// Destructor
DviStatusReport::~DviStatusReport() {
    // bf1

    // videoFormat
    org_himalay_ccif_VideoFormat__DELETE( videoFormat);
}

int DviStatusReport::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // bf1
    bf1= istream.readUI8();
    retVal += 1;
    // videoFormat
    retVal += videoFormat->read(istream);
    return retVal;
}

int DviStatusReport::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read bf1
    bf1= istream.readUI8();
    retVal += 1;
    // read videoFormat
    retVal += videoFormat->read(istream);

    return retVal;
}

int DviStatusReport::write(DataOutputStream& ostream) { // throws IOException

    {   /** fix dependent sizes for videoFormat **/
    }

    int retVal= 0;
    // write bf1
    ostream.writeUI8(bf1);
    retVal +=1;
    // write videoFormat
    if ( videoFormat != NULL ) {
        retVal +=videoFormat->write(ostream);
    }

    return retVal;
}

int DviStatusReport::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("DviStatusReport\n");
    dc.increaseIndent();
    int retVal= 0;
    // write bf1
    {
        dc.indent();
        dc.getPs().print("reserved: ");
        dc.getPs().printlnBin(getReserved(),3);
        dc.indent();
        dc.getPs().print("connectionStatus: ");
        dc.getPs().printlnBin(getConnectionStatus(),2);
        dc.indent();
        dc.getPs().print("hostHdcpStatus: ");
        dc.getPs().printlnBin(getHostHdcpStatus(),1);
        dc.indent();
        dc.getPs().print("deviceHdcpStatus: ");
        dc.getPs().printlnBin(getDeviceHdcpStatus(),2);
    }
    // write videoFormat
    if ( videoFormat != NULL ) {
        dc.indent();
        dc.getPs().println("videoFormat") ;
        retVal +=videoFormat->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for bf1
//ui8 DviStatusReport::getBf1()
//{
//return this->bf1;
//}

// Setter for bf1
//void DviStatusReport::setBf1(ui8 val)
//{
//this->bf1= val;
//}

// Getter for videoFormat
//org_himalay_ccif_VideoFormat__PTR_TYPE DviStatusReport::getVideoFormat()
//{
//return this->videoFormat;
//}

// Setter for videoFormat
//void DviStatusReport::setVideoFormat(org_himalay_ccif_VideoFormat__PTR_TYPE val)
//{
//this->videoFormat= val;
//}

int DviStatusReport::getReserved( ) {
    return ( this->bf1 & 0x000000e0 ) >> 5 ;
}

void DviStatusReport::setReserved(int val) {
    bf1=(  (this->bf1 & 0xffffff1f) | ( (val << 5 ) & 0x000000e0));
}

int DviStatusReport::getConnectionStatus( ) {
    return ( this->bf1 & 0x00000018 ) >> 3 ;
}

void DviStatusReport::setConnectionStatus(int val) {
    bf1=(  (this->bf1 & 0xffffffe7) | ( (val << 3 ) & 0x00000018));
}

int DviStatusReport::getHostHdcpStatus( ) {
    return ( this->bf1 & 0x00000004 ) >> 2 ;
}

void DviStatusReport::setHostHdcpStatus(int val) {
    bf1=(  (this->bf1 & 0xfffffffb) | ( (val << 2 ) & 0x00000004));
}

int DviStatusReport::getDeviceHdcpStatus( ) {
    return ( this->bf1 & 0x00000003 ) >> 0 ;
}

void DviStatusReport::setDeviceHdcpStatus(int val) {
    bf1=(  (this->bf1 & 0xfffffffc) | ( (val << 0 ) & 0x00000003));
}

int DviStatusReport::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_DviStatusReport__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::DviStatusReport* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::DviStatusReport* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code